src/models/room.js

```diff
@@ -207,7 +207,12 @@ Room.prototype.getLiveTimeline = function() {
  */
 Room.prototype.resetLiveTimeline = function(backPaginationToken) {
     for (let i = 0; i < this._timelineSets.length; i++) {
-        this._timelineSets[i].resetLiveTimeline(backPaginationToken);
+        // Flush out non-live timelines. We do this to reduce the amount of memory
+        // used, as storing multiple distinct copies of room state (each of which
+        // can be 10s of MBs) for each timeline is expensive. This is particularly
+        // noticeable when the client unsleeps and there are a lot of 'limited: true'
+        // responses. https://github.com/vector-im/riot-web/issues/3307#issuecomment-282895568
+        this._timelineSets[i].resetLiveTimeline(backPaginationToken, true);
```

Comment:
it shouldn't jump down if you're viewing historical events.

A handful of examples: what happens when you scroll down so that the historical timeline merges with the new one? what happens if you then scroll back up? what happens if you jump to a permalink to the events that are now in a timeline which is attached to the live timeline, but are no longer in `_eventIdToTimeline` ?

Mostly though, I think that a change that deliberately leaves the datastructures in an inconsistent state is a bad idea.